fastlane/spec/plugins_specs/plugin_info_collector_spec.rb

```diff
@@ -95,6 +103,13 @@
     it "handles plugin name with characters we don't want" do
       expect(collector.plugin_name_valid?('T!EST-na$me ple&ase')).to be_falsey
     end
+
+    it "detects if the plugin is already taken on RubyGems.org" do
+      stub_request(:get, "https://rubygems.org/api/v1/gems/already_taken.json").
+        with(headers: {'Accept' => '*/*', 'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent' => 'Ruby'}).
+        to_return(status: 200, body: {version: "1.0"}.to_json, headers: {})
```

Comment:
suggestion: define a helper method that accepts a `gem_name` and a boolean to control whether it is found, and sets up the stubbing, then reuse in all of these places.